src/main/java/apoc/export/graphml/XmlGraphMLReader.java

```diff
@@ -60,30 +62,82 @@ public XmlGraphMLReader reporter(Reporter reporter) {
             Object parse(String value) {
                 return Boolean.valueOf(value);
             }
+
+            Object parseList(String value) {
+                List parsedList = JsonUtil.parse(value, null, List.class);
+                return parsedList.toArray(new Boolean[parsedList.size()]);
+            }
         }, INT() {
             Object parse(String value) {
                 return Integer.parseInt(value);
             }
+
+            Object parseList(String value) {
+                List parsedList = JsonUtil.parse(value, null, List.class);
+                return parsedList.toArray(new Integer[parsedList.size()]);
+            }
         }, LONG() {
             Object parse(String value) {
                 return Long.parseLong(value);
             }
+
+            Object parseList(String value) {
+                List parsedList = JsonUtil.parse(value, null, List.class);
+                Long[] result = new Long[parsedList.size()];
+                for (int i = 0; i < parsedList.size(); i++) {
+                    Object item = parsedList.get(i);
+                    if (item instanceof Long) {
```

Comment:
why always longs? shouldn't we look into the datatypes of the parsed list?
also ((Number)item).longValue()

Reply:
also in the other places

Reply:
Longs because the list was defined as a list of longs. I'm not sure if the input format makes a difference to how neo4j stores them, or if it can handle different types of numbers in the same array, so I can switch it to whatever you think is best.

Reply:
I tried converting to a Number[] for the Long/Int/Double/Float methods and it looks like that's not supported:

`java.lang.IllegalArgumentException: java.lang.Number[] is not a supported property value type`

So we'll have to have some kind of type conversion to a specific number type. Although I think it could be possible with reflection to figure out a type based on the input values, forcing them into the type that was declared in the graphml is simpler and probably